src/de/jost_net/JVerein/io/BeitragsUtil.java

```diff
@@ -43,10 +45,69 @@ public static double getBeitrag(Beitragsmodel bm, Zahlungstermin zt, int zr,
     switch (bm)
     {
       case GLEICHERTERMINFUERALLE:
-        betr = bg.getBetrag();
+        if(Einstellungen.getEinstellung().getGeburtsdatumPflicht() && bg.getHasAltersstaffel())
+        {
+          Einstellungen.getEinstellung().getAltersModel();
+          String stufen = Einstellungen.getEinstellung().getBeitragAltersstufen();
+          if(stufen != null && stufen != "")
+          {
+            AltersgruppenParser ap = new AltersgruppenParser(stufen);
+            int i = 0;
+            int nummer = -1;
+            while(ap.hasNext())
+            {
+              VonBis vb = ap.getNext();
+              if(alter >= vb.getVon() && alter <= vb.getBis())
```

Comment:
Ich bekam hier eine NullPointer exception weil mein Mitglied jünger als ein Jahr war. Da liefert getAlter() eine null zurück.
Die Meldung war "Zahlungsinformationen bei ...". Das hatte mir nicht weiter geholfen. Durch debugging habe ich es gefunden.
Das kann auch passieren wenn das Geburtdatum fehlt weil z.B. der Schalter "Geburtsdatum Pflicht" erst eingeschaltet wurde wenn schon Mitglieder ohne Geburtsdatum existieren.
Evtl. sollte man in dieser Methode auch alter auf null prüfen und einen passenderen Fehlertext ausgeben z.B. "Geburtsdatum fehlt oder Mitglied jünger als 1 Jahr bei ...".

Allerdings stellt sich mir die Frage warum man bei Alter <1 in der Klasse Datum Zeile 193 eine null zurück gibt. Man könnte doch auch die 0 zurückgeben und null nur bei <0. Aber vielleicht gibt es wo anders Probleme mit Alter 0. Wahrscheinlich wird es aber wohl auch keine Mitglieder geben die noch Babys sind.

Reply:
die Berechnung des Alters habe ich korrigiert. 
Allerdings weiß ich noch nicht wie ich die Fehlermeldung ändern kann. In AbrechnungSEPA wird eine NPE aufgefangen. Um zu prüfen warum müsste ich hier die Tests ob es eine Altersstaffel gibt und ein Geburtsdatum existier machen. Das finde ich aber nicht schön, da es eigentlich nur ins BeitragsUtil gehört. Da habe ich jedoch keinen Zugriff auf den Namen des Mitglieds um ihn mit auszugeben.
Hast du eine Idee wie wir das lösen können?

Reply:
Statt eintritt, austritt und alter könnte man direkt das Mitglied an BeitragsUtil übergeben. Dann hätte man auch Zugriff auf Namen und Vornamen.

Reply:
habe ich so gelöst